x-pack/legacy/plugins/apm/public/components/app/TraceLink/index.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import url from 'url';
+import styled from 'styled-components';
+import { Redirect } from 'react-router-dom';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { QueryParams } from '../../shared/Links/apm/ExternalLinks';
+
+const CentralizedContainer = styled.div`
+  height: 100%;
+  display: flex;
+`;
+
+const redirectToTransactionDetailPage = (
+  transaction: Transaction,
+  { rangeFrom, rangeTo }: Partial<QueryParams>
```

Comment:
To make this required you can either use `Pick` and `Required`. Or just defined it manually (without using `QueryParams`)

Reply:
I'm actually making them Optional. I want to use QueryParams, because it will use the same type defined in `ExternalLinks.ts`.